Спринт 13.

sds96 opened the pull request:
Экран аудиоплеера при нажатии на трек в поиске.

sds96 committed:
Спринт 13.
Экран аудиоплеера при нажатии на трек в поиске.

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/AudioPlayerActivity.kt:
```diff
@@ -0,0 +1,58 @@
+        }
+
+        val track : Track? = if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU)
+            intent.getSerializableExtra("track", Track::class.java)
```
Ключи для данных лучше выносить в константы, тем более когда они используются больше, чем в одном месте

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/Track.kt:
```diff
@@ -1,9 +1,18 @@
     val trackTimeMillis: Int = 0
-)
\ No newline at end of file
+) : Serializable
```
Лучше использовать интерфейс `Parcelable`, т.к. в отличии от `Serializable` он не использует рефлексию. К тому же `Parcelable` является рекомендуемым способом передачи данных между компонентами в Android и он активно поддерживается гуглом в сдк. Для удобства есть плагин `kotlin-parcelize`, который позволит сгенерировать реализацию этого интерфейса

ArturNurtdinov commented on app/src/main/res/layout/activity_audio_player.xml:
```diff
@@ -0,0 +1,238 @@
+    <ImageButton
+        android:id="@+id/backButton"
+        android:layout_width="48dp"
+        android:layout_height="48dp"
```
Одинаковые размеры - в dimens

ArturNurtdinov commented on app/src/main/res/layout/activity_audio_player.xml:
```diff
@@ -0,0 +1,238 @@
+        android:layout_height="wrap_content"
+        app:layout_constraintBaseline_toBaselineOf="@id/trackDurationLine"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="@id/leftTextBarrier"
```
Круто, что не забыл подумать о том, чтобы текст корректно обрезался, если вдруг он будет длинным. Решение с барьером - огонь!

sds96 committed:
Правка по ревью.
Замена Serializable на Parcelable, ключ в константу, одинаковые размеры в dimens.
